src/domain/email/email.repository.ts

```diff
@@ -149,6 +150,33 @@ export class EmailRepository implements IEmailRepository {
     return this.emailDataSource.deleteEmail(args);
   }
 
+  async updateEmail(args: {
+    chainId: string;
+    safeAddress: string;
+    emailAddress: string;
+    account: string;
+  }): Promise<void> {
+    const email = new EmailAddress(args.emailAddress);
+    const verificationCode = this._generateCode();
+
+    try {
+      await this.emailDataSource.updateEmail({
```

Comment:
What if the email address that we are updating is the same.

E.g.: A verified `foo@example.com` is updated with `foo@example.com`. This action should probably result in a client error to signal that the same email was provided, and we don't update it on our side (as it stands, updating it to the same value would result in setting a verified email as unverified).